File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/05_While_Loop_Exercise/02_Exam_Preparation.js
function examPreparation(input) {  // ["3","Money","6","Story","4","Spring Time","5","Bus","6","Enough"]);
    // let exitMessage = "Enough";
    // let badGrade = 4;

    let badGradesLimit = Number(input[0]);
    let badGradesCurrent = 0;
    let totalTasks = 0;
    let sumTaks = 0;
    let lastProblemName = '';

    let i = 1;
    while (true) {
        let taskName = input[i];
        i++;

        if (taskName == "Enough") {  // taskName == "Enough"
            console.log(`Average score: ${(sumTaks / totalTasks).toFixed(2)}`);  // avg = (a1 + a2 + a2) / totalTasks 
            console.log(`Number of problems: ${totalTasks}`);
            console.log(`Last problem: ${lastProblemName}`);
            break;
        }


        let taskGrage = Number(input[i]);
        i++;


        totalTasks++;
        sumTaks += taskGrage;
        lastProblemName = taskName;

        if (taskGrage <= 4) {  // taskGrage <= 4
            badGradesCurrent++;
            if (badGradesCurrent == badGradesLimit) {
                console.log(`You need a break, ${badGradesCurrent} poor grades.`);
                break;
            }

        }
    }

}


// examPreparation(["3",
//     "Money",
//     "6",
//     "Story",
//     "4",
//     "Spring Time",
//     "5",
//     "Bus",
//     "6",
//     "Enough"]);

// examPreparation(["2",
//     "Income",
//     "3",
//     "Game Info",
//     "6",
//     "Best Player",
//     "4"]);

File: Python_Courses/Python_Basics/Preparation/05_While_Loop_Exercise/02_Exam_Preparation.py
POOR_GRADE_THRESHOLD = 4

failed_count = 0
total_grades = 0
problems_count = 0
last_problem_name = ""
is_failed = False

failed_count_boundary = int(input())

while True:
    problem_name = input()
    if problem_name == "Enough":
        break

    last_problem_name = problem_name
    problem_score = int(input())

    total_grades += problem_score
    problems_count += 1

    if problem_score <= POOR_GRADE_THRESHOLD:
        failed_count += 1
        if failed_count >= failed_count_boundary:
            is_failed = True
            break

if not is_failed:
    avg_grade = total_grades / problems_count
    print(f"Average score: {avg_grade :.2f}")
    print(f"Number of problems: {problems_count}")
    print(f"Last problem: {last_problem_name}")
else:
    print(f"You need a break, {failed_count} poor grades.")
